src/charactersheet/components/flip-card/index.js

```diff
@@ -0,0 +1,202 @@
+        // apply bindings to trigger resize
+        var innerBindingContext = ko.bindingEvent.startPossiblyAsyncContentBinding(element, bindingContext);
+        ko.applyBindingsToDescendants(innerBindingContext, element);
+
+        var value = valueAccessor();
+        var hiddenCallback = ko.utils.unwrapObservable(value.hiddenCallback);
+        var shownCallback = ko.utils.unwrapObservable(value.shownCallback);
+
+        if (shownCallback) {
+             // Register callbacks.
+            $(element).on('show.bs.collapse', shownCallback);
+            $(element).on('shown.bs.collapse', shownCallback);
+        }
+        if (hiddenCallback) {
+             // Register callbacks.
+            $(element).on('hidden.bs.collapse', hiddenCallback);
+        }
+        return { controlsDescendantBindings: true };
+    },
+    // eslint-disable-next-line no-unused-vars
+    update: function(element, valueAccessor, allBindingsAccessor) {
+    }
+};
+
+export class FlipCardComponentViewModel {
+    constructor(params) {
+        // Unique id for the card's data
+        this.dataId = ko.utils.unwrapObservable(params.dataId) || '1';
+        // element id for collapsable behavior
+        this.paramElementId = ko.utils.unwrapObservable(params.elementId);
+        // unique identifier for this card
+        this.elementId = `${this.paramElementId}_${this.dataId}`;
+
+        this.tabId = ko.utils.unwrapObservable(params.tabId);
+
+        // Whether or not to trigger animations on collapse
+        this.collapsable = ko.observable(ko.utils.unwrapObservable(params.collapsable) || false);
+
+        // contextual data to be passed to children objects
+        this.context = params.context;
+
+        // Whether or not this 'back' is displayed.
+        this.showBack = ko.observable(false);
+
+        // use a custom parameter
+        if (params.showBack !== undefined) {
+            this.showBack = params.showBack;
+        }
+
+        if (params.onFlip) {
+            // callback to trigger on flip
+            this.onFlip = params.onFlip;
+        }
+
+        if (params.onResize) {
+            // callback on
+            this.onResize = params.onResize;
+        }
+        let initialDefaultHeight = 350;
+        if (this.collapsable()) {
+            initialDefaultHeight = 0;
+        }
+        // calculated element height
+        this.elementMeasure = ko.observable(initialDefaultHeight).extend({ deferred: true });
+        if (params.defaultHeight) {
+            const paramHeight = parseInt(ko.utils.unwrapObservable(params.defaultHeight));
+            if (Number.isInteger(paramHeight)) {
+                this.elementMeasure(paramHeight);
+            }
+        }
+        this.elementHeight = ko.pureComputed(()=> {
+            return `${this.elementMeasure()}px`;
+        }).extend({ deferred: true });
+    }
+
+    showFront = ko.pureComputed(() => !this.showBack());
+
+    toggleMode = (data, event) => {
+        event.preventDefault();
```

Comment:
`event` is undefined here and causes an issue when trying to use anything with a `flip-card`.
I've commented it out for local testing.

Reply:
I think that shouldn't be there. I'll remove it.